Modules/Module_Processing/Module_ASL/xASL_wrp_PreparePV.m

```diff
@@ -143,22 +143,54 @@
 end
 
 %% ------------------------------------------------------------------------------------------
-% 6     Prepare pGM and pWM in the ASL native space
-% by presmoothing before downsampling to ASL space
+% 6     Prepare pGM, pWM, pCSF, WMH_Segm, and also Lesion and ROI maps in the ASL native space
+
+% We need to take into account the resolution difference between T1 and ASL.
+% The correct downsampling is done by first presmoothing the image and then reslicing using normal SPM routines.
+
+% Transform GM and WM segmentation to the ASL space
 xASL_im_PreSmooth(x.P.Path_PWI,x.P.Path_c1T1, x.P.Path_PVgm,x.S.optimFWHM_Res_mm,[],x.P.Path_mean_PWI_Clipped_sn_mat, 1);
 xASL_im_PreSmooth(x.P.Path_PWI,x.P.Path_c2T1, x.P.Path_PVwm,x.S.optimFWHM_Res_mm,[],x.P.Path_mean_PWI_Clipped_sn_mat, 1);
 xASL_spm_reslice(x.P.Path_PWI, x.P.Path_PVgm, x.P.Path_mean_PWI_Clipped_sn_mat, 1, x.settings.Quality, x.P.Path_PVgm);
 xASL_spm_reslice(x.P.Path_PWI, x.P.Path_PVwm, x.P.Path_mean_PWI_Clipped_sn_mat, 1, x.settings.Quality, x.P.Path_PVwm);
 
-if xASL_exist(x.P.Path_c3T1) % for backward compatibility, where c3T1 wasnt always available
+% Transform CSF segmentation to the ASL space
+if xASL_exist(x.P.Path_c3T1) % Check file existence as for older versions c3T1 was not always available
 	xASL_im_PreSmooth(x.P.Path_PWI,x.P.Path_c3T1,x.P.Path_PVcsf,x.S.optimFWHM_Res_mm,[],x.P.Path_mean_PWI_Clipped_sn_mat, 1);
 	xASL_spm_reslice(x.P.Path_PWI, x.P.Path_PVcsf, x.P.Path_mean_PWI_Clipped_sn_mat, 1, x.settings.Quality, x.P.Path_PVcsf);
 end
 
+% Transform WMH segmentation maps to the ASL space
 if xASL_exist(x.P.Path_WMH_SEGM) 
 	xASL_im_PreSmooth(x.P.Path_PWI,x.P.Path_WMH_SEGM, x.P.Path_PVwmh,x.S.optimFWHM_Res_mm,[],x.P.Path_mean_PWI_Clipped_sn_mat, 1);
 	xASL_spm_reslice(x.P.Path_PWI, x.P.Path_PVwmh, x.P.Path_mean_PWI_Clipped_sn_mat, 1, x.settings.Quality, x.P.Path_PVwmh);
 end
 
+% Transform Lesion and ROI masks to the ASL space. Note that we transform binary masks to maps with values between 0 and 1 so that the PV factor is recorded.
+% The maps have then have to be binarized again with a given threshold before using them again as masks

```

Comment:
Currently nowhere, because we are not yet doing native-space analysis with ROIs and Lesions -> this is currently missing and has to be implemented. But rather than doing the thresholding here, it might be wiser to do it directly in the population moduel as we can nicely adapt this and change this later or even make it somehow variable. 
Haha - of course not leaving that to the user. I'm just leaving the question of correct threshold for the population module. This comment is not for the user but for our record.

Fully agree about probabilistic/binary. But since I call this PVLesion_T1_1, it makes it clear in the name that this is probabilistic rather than binary. 

Yes - in ideal world, we don't change probabilistic to binary and vice versa. But tumor are rather tricky and I don't want to make a clear decision on how to threshold it. It would be wrong to cut it at 50% as we can miss a large part of the tumor as these tend to be very thin at certain times. So this decision should be made in the population module and we either consider the PV or we threshold.

Reply:
I would argue why the ASL native space lesion mask needs to be implemented. "Because Ivar asks for it" is not a good argument. Otherwise, I see your point.

Reply:
we wanted to have native space analysis and that has to be complete. It's only that I forgot about this option when doing native space analysis. And only Ivar actually discovered this shortcoming. So it's not adding that because he wanted it. But it's somethign that should have been there from the get go, but it wasn't and he discovered that.

Reply:
You know my opinion ;)

I have now fixed this by adding many warnings and making the code clear that this needs fixing when we fix the whole native space mess.